dvc/objects/db/base.py

```diff
@@ -75,6 +76,40 @@ def get(self, hash_info: "HashInfo"):
             hash_info,
         )
 
+    def _add_file(self, from_fs, from_info, to_info, move):
+        self.makedirs(to_info.parent)
+        use_move = isinstance(from_fs, type(self.fs)) and move
+        try:
+            if use_move:
+                self.fs.move(from_info, to_info)
+            elif isinstance(from_fs, LocalFileSystem):
+                if not isinstance(from_info, from_fs.PATH_CLS):
+                    from_info = from_fs.PATH_CLS(from_info)
+                self.fs.upload(from_info, to_info)
+            elif isinstance(self.fs, LocalFileSystem):
+                from_fs.download_file(from_info, to_info)
```

Comment:
Btw, should both of these use `_file` methods?

Reply:
What do you mean by that?

Reply:
@isidentical I see that we use `fs.upload` but `fs.download_file`. Wondering if they both should be `upload/download` or `upload_file/download_file` (likely the latter if we take fsspec's `get/put_file` future in the context, but maybe I'm missing something important here)

Reply:
We don't have `upload_file()`, so the `upload()` is `put_file`. And `download_file()` is the `get_file()`. The reason that I didn't use `download()` is that, it has an extra check against whether the source path is a directory, which is an waste of API call for this case where we know we are adding a single entity (file).

Reply:
Makes sense, thank you! I guess we'll switch to get_file/put_file when reducing fsspec_wrapper in the future.